src/ComposerRequireChecker/ASTLocator/LocateASTFromFiles.php

```diff
@@ -27,7 +27,7 @@ public function __construct(Parser $parser, ?ErrorHandler $errorHandler)
     /**
      * @param Traversable $files
      *
-     * @return Traversable|array[] a series of AST roots, one for each given file
+     * @return Traversable a series of AST roots, one for each given file
      */
     public function __invoke(Traversable $files): Traversable
```

Comment:
We can probably switch to `iterable` /cc @maglnet

Reply:
@weirdan the point of `array[]` is to indicate that each element of `$files` is an `array` - the change is incorrect here

Reply:
oh, so that's phpstorm-style generics then, and not a union type? Ok.

Reply:
`array[]` is `array<array>`, basically

Reply:
It's not an array though, as it's used in `file_get_contents()` (thus it's a string). So as far as I understand this should be `Traversable|string[]` in this case (based on [this comment](https://intellij-support.jetbrains.com/hc/en-us/community/posts/115000143944-PHPDoc-collection-generics-notation-supported-?page=1#community_comment_115000171490))

Reply:
@weirdan needs adjustment then, but not removal :+1: